### DISCUSSION TITLE
How to avoid the Auto arrange every time while adding new data in Existing organic layout diagram.

### DISCUSSION THREAD

**Discussion by venkatN467** [maxGraph/maxGraph#518] [Q&A]
Earlier we are using MX graph (mxFastOrganicLayout) in my application. but we are noticing each and every time while adding new data or new relation. entire existing structure/format is rearranging (Auto arranging) using organic layout. so, this will be avoided with MAXGraph .

**Comment by tbouffard** [comment]
Hi,
Can you provide some code to reproduce what you see (you may adapt https://github.com/maxGraph/maxGraph/blob/7f4d5a4042c33141f1ef40a5618698aaec657ca0/packages/html/stories/HierarchicalLayout.stories.ts)? for now, i don't understand what you are describing.
Are you seeing the same behavior with both mxGraph and maxGraph or only with maxGraph?

**Reply by venkatN467** [reply]
Hi ,

Thanks for your Reply. 

We are yet to incorporate the Max Graph, for now just wanted to take below example and let's say if I add one more alphabet/circle in the existing Diagram. then we wanted to know if the entire layout will change its existing position, or it will just add one alphabet/circle ?
![OrganicLayout](https://github.com/user-attachments/assets/a7ad2ed0-030c-453e-8396-d62d7f5fdeee)

**Reply by tbouffard** [reply]
AFAIK, maxGraph behaves as mxGraph because no functional change has been done on the layout implementation.
In https://maxgraph.github.io/maxGraph/demo/?path=/story/layouts-autolayout--default, cells are dynamically added and there is specific code to relayout the graph with an animation each time a cell is added.
Another example is available in https://github.com/maxGraph/maxGraph/blob/e1622a07da1785ed1e2b3486e98b71c2ab4fe421/packages/core/src/view/layout/ParallelEdgeLayout.ts#L38-L60

You could also modify the example you use to create the screenshot in your previous comment: add a button that will add a vertex with an edge on the graph. You will then now if the layout is automatically executed.

You can also check the `LayoutManager` class. Its JSDoc mentions what you seem to look for: _Implements a layout manager that runs a given layout after any changes to the graph_
